### ISSUE TITLE
TextureLayerd set_layer_data silently fails if the input image is a different size. 

### ISSUE THREAD

**Issue by huttyblue** [godotengine/godot#56488]
### Godot version

3.4.1.rc.mono

### System information

Windows 10, nvidia gtx970, gles3

### Issue description

If you try to use set_layer_data on a texture3D and pass it an image of a different size the end result when getting the data back will be a blank image, all pixels (0,0,0,0). No errors in either console, no crash, just a silent fail. 

Specifically I was trying to pass an image of size 1024x1024 to a tex3d of size 512x512, with 64 layers. 

Once I changed the size of the tex3D to match the image it started correctly setting the layer data to match the image sent in.

### Steps to reproduce

Create a Texture3d of size 512x512x64
Create an image of size 1024x1024
Run set_layer_data  on the Texture3d passing in your image to layer 2.
Run get_layer_data on layer 2 and print the results of a get_pixel.
The results will always be color(0,0,0,0)
The console will have no errors to indicate something is wrong.

**Comment by Calinou** [comment]
@huttyblue Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by miskatonicstudio** [comment]
Just checked the `3.4.1-stable` branch and confirmed:
```
	var t = Texture3D.new()
	t.create(512, 512, 4, Image.FORMAT_RGBA8, 0)
	var i = Image.new()
	i.create(1024, 1024, false, Image.FORMAT_RGBA8)
	i.fill(Color.blue)
	print(i.get_data()[0], " ", i.get_data()[1], " ", i.get_data()[2], " ")
	t.set_layer_data(i, 2)
	var ld = t.get_layer_data(2)
	print(ld.get_data()[0], " ", ld.get_data()[1], " ", ld.get_data()[2], " ")
```
returns:
```
0 0 255 
0 0 0 
```

However, it seems to work if the texture size is larger than the image size:
```
	var t = Texture3D.new()
	t.create(512, 512, 4, Image.FORMAT_RGBA8, 0)
	var i = Image.new()
	i.create(256, 256, false, Image.FORMAT_RGBA8)
	i.fill(Color.blue)
	print(i.get_data()[0], " ", i.get_data()[1], " ", i.get_data()[2], " ")
	t.set_layer_data(i, 2)
	var ld = t.get_layer_data(2)
	print(ld.get_data()[0], " ", ld.get_data()[1], " ", ld.get_data()[2], " ")
```
returns:
```
0 0 255 
0 0 255
```

The simplest solution would be to add another `ERR_FAIL_COND` if the image and texture size are different. Another solution is to scale the image both ways.

**Comment by Calinou** [comment]
> The simplest solution would be to add another ERR_FAIL_COND if the image and texture size are different.

I think this is the preferred solution, it's be more consistent with what's done in the Image class.

**Comment by miskatonicstudio** [comment]
I've created a PR to `3.4`. The structure in `master` differs quite a bit, I didn't find `set_layer_data`. If there is another place I should update there, please let me know ;)

**Comment by akien-mga** [comment]
Fixed by #56679.
